service-api/module/Application/test/ApplicationTest/Services/DWP/DwpApi/DwpApiServiceTest.php

```diff
@@ -0,0 +1,461 @@
+        CitizenResponseDTO $citizenResponseDTO,
+        DetailsResponseDTO $detailsResponseDTO,
+    ): void {
+        $this->assertEquals(
+            $expected,
+            $this->dwpApiService->compareRecords(
+                $caseData,
+                $detailsResponseDTO,
+                $citizenResponseDTO,
+                self::NINO
+            )
+        );
+    }
+
+    public static function compareRecordsData(): array
+    {
+        $successMatchDTO = new CitizenResponseDTO([
+            "jsonapi" => [
+                "version" => "1.0"
+            ],
+            "data" => [
+                "id" => "be62ed49-5407-4023-844c-97159ec80411",
+                "type" => "MatchResult",
+                "attributes" => [
+                    "matchingScenario" => "Matched on NINO"
+                ]
+            ]
+        ]);
+
+        $noMatchDTO = new CitizenResponseDTO([
+            "jsonapi" => [
+                "version" => "1.0"
+            ],
+            "data" => [
+                "id" => "",
+                "type" => "",
+                "attributes" => [
+                    "matchingScenario" => ""
+                ]
+            ]
+        ]);
+
+        $successDetailsDTO = new DetailsResponseDTO(static::DETAILS_RESPONSE);
+
+        $failDetailsResponse = static::DETAILS_RESPONSE;
+        $failDetailsResponse['data']['attributes']['identityVerificationStatus'] = '';
+
+        $failDetailsDTO = new DetailsResponseDTO($failDetailsResponse);
+
+        $caseData = CaseData::fromArray(static::CASE);
+
+        return [
+            [
+                true,
+                $caseData,
+                $successMatchDTO,
+                $successDetailsDTO
+            ],
+            [
+                false,
+                $caseData,
+                $noMatchDTO,
+                $failDetailsDTO
+            ],
+            [
+                false,
+                $caseData,
+                $noMatchDTO,
+                $successDetailsDTO
+            ],
+        ];
+    }
+
+    public function testValidateNino(): void
+    {
+        $successMockResponseMatchData = [
+            "jsonapi" => [
+                "version" => "1.0"
+            ],
+            "data" => [
+                "id" => "be62ed49-5407-4023-844c-97159ec80411",
+                "type" => "MatchResult",
+                "attributes" => [
+                    "matchingScenario" => "Matched on NINO"
+                ]
+            ]
+        ];
+        $successMockResponseDetailsData = static::DETAILS_RESPONSE;
+
+        $successMock = new MockHandler([
+            new GuzzleResponse(200, [], json_encode($successMockResponseMatchData, JSON_THROW_ON_ERROR)),
+            new GuzzleResponse(200, [], json_encode($successMockResponseDetailsData, JSON_THROW_ON_ERROR)),
+        ]);
+        $handlerStack = HandlerStack::create($successMock);
+        $successClient = new Client(['handler' => $handlerStack]);
```

Comment:
Reflecting on this, this test would probably be better if it used a PHPUnit mock rather than a mock handler so that we could check that the correct endpoints are called. At the moment, it just tests that two HTTP requests are made but we don't check the path, headers or body being used.